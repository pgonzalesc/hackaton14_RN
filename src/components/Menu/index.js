import React, {useState} from 'react';
import {StyleSheet, View, TouchableHighlight, TouchableOpacity, Text} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const Menu = ()=>{
    const [selectedOption, setSelectedOption] = useState({coffee:true, cake:false, snack:false});

    const handleOption = (option) => {
        if(option === 'coffee')
            setSelectedOption({coffee:true, cake:false, snack:false});
        else if(option === 'cake')
            setSelectedOption({coffee:false, cake:true, snack:false});
        else if(option === 'snack')
            setSelectedOption({coffee:false, cake:false, snack:true});
    }

    return (
        <View style={styles.container}>
            <View style={styles.containerButton}>
                {selectedOption.coffee 
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleOption('coffee')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-cafe-outline'} size={20}/>
                                </View>
                                <Text style={styles.textoSelected}>Coffee</Text>
                            </View>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleOption('coffee')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-cafe-outline'} size={20}/>
                                </View>
                                <Text style={styles.texto}>Coffee</Text>
                            </View>
                        </TouchableOpacity>
                }
            </View>
            <View style={styles.containerButton}>
                {selectedOption.cake 
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleOption('cake')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-ice-cream-outline'} size={20}/>
                                </View>
                                <Text style={styles.textoSelected}>Cake</Text>
                            </View>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleOption('cake')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-ice-cream-outline'} size={20}/>
                                </View>
                                <Text style={styles.texto}>Cake</Text>
                            </View>
                        </TouchableOpacity>
                }
            </View>
            <View style={styles.containerButton}>
                {selectedOption.snack 
                     ?  <TouchableOpacity style={styles.buttonSelected} onPress={()=>handleOption('snack')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-fast-food-outline'} size={20}/>
                                </View>
                                <Text style={styles.textoSelected}>Snack</Text>
                            </View>
                        </TouchableOpacity>
                     :  <TouchableOpacity style={styles.button} onPress={()=>handleOption('snack')}>
                            <View style={{flexDirection:"row", alignItems:"center"}}>
                                <View style={styles.icono}>
                                    <Icon name={'md-fast-food-outline'} size={20}/>
                                </View>
                                <Text style={styles.texto}>Snack</Text>
                            </View>
                        </TouchableOpacity>
                }
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
      //flex: 1,
      flexDirection: "row",
    },
    title: {
      fontSize: 20,
      marginTop: 10,
      marginHorizontal: 20,
      fontWeight: 'bold',
    },
    containerButton:{
        padding: 10,
        margin: 10,
    },
    button:{
        backgroundColor: '#e1e1e1',
        paddingHorizontal: 15,
        paddingVertical: 10,
        borderRadius: 25,
    },
    buttonSelected: {
        backgroundColor: '#308278',
        paddingHorizontal: 15,
        paddingVertical: 10,
        borderRadius: 25,
    },
    icono:{
        backgroundColor:'white', 
        marginRight: 10, 
        padding: 5, 
        borderRadius: 15,
    },
    texto: {
        color:'#000',
        fontWeight: 'bold',
    },
    textoSelected: {
        color:'#FFF',
        fontWeight: 'bold',
    }
  });

export default Menu;