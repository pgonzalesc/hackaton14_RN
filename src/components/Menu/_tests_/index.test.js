import React from 'react';
import { TouchableOpacity } from 'react-native';
import {shallow} from 'enzyme';
import Menu from '../index';
const onPress = jest.fn();
describe('Pruebas en <Menu/>', () => {
    let wrapper = shallow(<Menu/>);

    beforeEach(()=>{
        let wrapper = shallow(<Menu/>);
    });

    test('Debe de mostrar <Menu/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe seleccionar Coffee', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(0).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(0).children().children().at(1).children().text();
        expect(value).toBe('Coffee');
    });

    test('Debe seleccionar Cake', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(1).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(1).children().children().at(1).children().text();
        expect(value).toBe('Cake');
    });

    test('Debe seleccionar Snack', () => {
        expect(onPress).not.toHaveBeenCalled();
        wrapper.find(TouchableOpacity).at(2).simulate('press');
        const value = wrapper.find(TouchableOpacity).at(2).children().children().at(1).children().text();
        expect(value).toBe('Snack');
    });
});
