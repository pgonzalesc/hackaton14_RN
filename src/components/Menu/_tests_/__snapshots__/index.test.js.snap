// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Menu/> Debe de mostrar <Menu/> correctamente 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#308278",
          "borderRadius": 25,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 15,
              "marginRight": 10,
              "padding": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="md-cafe-outline"
            size={20}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontWeight": "bold",
            }
          }
        >
          Coffee
        </Text>
      </View>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#e1e1e1",
          "borderRadius": 25,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 15,
              "marginRight": 10,
              "padding": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="md-ice-cream-outline"
            size={20}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
              "fontWeight": "bold",
            }
          }
        >
          Cake
        </Text>
      </View>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#e1e1e1",
          "borderRadius": 25,
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 15,
              "marginRight": 10,
              "padding": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="md-fast-food-outline"
            size={20}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
              "fontWeight": "bold",
            }
          }
        >
          Snack
        </Text>
      </View>
    </ForwardRef>
  </View>
</View>
`;
